packages/core-data/src/test/resolvers.js

```diff
@@ -115,6 +111,31 @@ describe( 'getEntityRecord', () => {
 			1
 		);
 	} );
+
+	it( 'does not make a request if the record is already in store', async () => {
+		const select = {
+			hasEntityRecord: jest.fn( () => true ),
+		};
+
+		await getEntityRecord( 'root', 'postType', 'post', {
+			context: 'edit',
+			_fields: 'id,title,slug',
+		} )( { dispatch, select, registry, resolveSelect } );
+
+		// No request should have been made.
+		expect( triggerFetch ).not.toHaveBeenCalled();
```

Comment:
This looks like a better candidate for integration tests from https://github.com/WordPress/gutenberg/pull/71474.

The current test gives us a false sense of safety; even if the `hasEntityRecord` behavior changes, the test will still pass.